contracts/Kasu.sol

```diff
@@ -121,16 +121,27 @@ contract Kasu is KasuStorage, KasuMath {
 
     // [Feature 3] Borrower's dashboard
     // borrower can see all the NFTs they borrowed
-    function viewRentedListings(address borrowerAddress) public view returns (Listing[] memory){
-        uint256[] memory listingIds = _getListingIds();
-        Listing[] memory listings = new Listing[](listingIds.length);
-        for (uint i = 0; i < listingIds.length; i++) {
-            Listing memory listing = _getListingById(listingIds[i]);
-            if (listing.rental.borrowerAddress == borrowerAddress){
-                listings[i] = listing;
+    function viewRentedListings() public view returns (Listing[] memory){
+        // Note: Dynamic arrays are not allowed unless they are storage
+        // so we need to allocate a statically sized-array to return.
+        // We can probably optimize this, but this should functionally work for now.
+        uint listingsRented = 0;
```

Comment:
also @amarjandu it looks like you're filtering bogus 0x data from the front-end. This is b/c you allocated a larger than needed array and there are certain slots that are not filled out (not all rented listings are the user's). To solve this you need to only allocate the correct size array and put the correct number of elements in it. Also we probably want the output of this function to be predictable so that we can easily test this.